<launch>
    <arg name="rviz_file" default="$(find youbot_localization)/config/localization.rviz"/>

    <arg name="node_namespace" default="apriltags2_ros_continuous_node" />
    <arg name="camera_name" default="ids_xs_camera_L" />
    <arg name="camera_frame" default="$(arg camera_name)" />
    <arg name="image_topic" default="image_rect" />

    <group ns="$(arg camera_frame)">
        <param name="tf_prefix" value="$(arg camera_frame)" />
        <!-- Set parameters -->
        <rosparam command="load" file="$(find youbot_localization)/config/apriltags/settings.yaml" ns="$(arg node_namespace)" />
        <rosparam command="load" file="$(find youbot_localization)/config/apriltags/tags.yaml" ns="$(arg node_namespace)" />
        
        <!-- TODO: in future map frame creation by tags -->
        <!--node pkg="tf" type="static_transform_publisher" name="map_to_camera_tf"
            args="0 0 1.0 0 0 -1.57 map camera 100"/-->

        <!--node pkg="nodelet" type="nodelet" name="camera_process2" args="manager" output="screen"/-->
        <!--node pkg="nodelet" type="nodelet" name="$(arg node_namespace)"
            args="load apriltags2_ros/ContinuousDetector /camera_process" output="screen"-->
        <node pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="$(arg node_namespace)" clear_params="true" output="screen">
            <!-- Remap topics from those used in code to those on the ROS network-->

            <param name="camera_frame" type="str" value="$(arg camera_frame)" />
            <param name="publish_tag_detections_image" type="bool" value="true" />      <!-- default: false -->
        </node>

        <!-- Run additional localization service -->
        <node pkg="youbot_localization" type="tags_tf_to_robot_pos_node" name="tf_to_pos_node">
            <param name="map_frame" type="str" value="/map"/>
            <param name="robot_frame" type="str" value="robot_2"/>
            <param name="publish_frequency" value="10"/>
            <param name="is_stamped" type="bool" value="true"/>
        </node>

    </group>

    <node pkg="rviz" type="rviz" name="rviz" args="-d $(arg rviz_file)"/>
</launch>
